service/lib/agama/storage/configs/search.rb

```diff
@@ -32,14 +32,26 @@ def self.new_for_search_all
           new.tap { |c| c.if_not_found = :skip }
         end
 
+        # Search by name.
```

Comment:
This design of `Configs::Search` with separate explicit attributes for name, size and partition_number doesn't look too flexible for the future. How will it work in the future to combine conditions? I mean stuff like this (not sure about the syntax yet):

```json
{
  "partitions": [
    {
      "search": {
        "if_not_found": "create",
        "condition": {
          { "or": [
              { "number": 2 },
              { "and": [
                  { "size": { "greater": "1 MiB" } },
                  { "size": { "less": "10 MiB" } }
                ]
              }
            ]
          }
        }
      }
    }
  ] 
}
```

Reply:
I did the minimal changes to `Configs::Search` to support the current needs. It will evolve according to what we want to offer. I think in the fututure we will have somehing like `Configs#Search#condition` and `Configs#SearchCondition`, or something similar.